v2/inbound_middleware.go

```diff
@@ -35,42 +35,93 @@ import (
 // UnaryInboundTransportMiddleware is re-used across requests and MAY be called multiple
 // times for the same request.
 type UnaryInboundTransportMiddleware interface {
-	Handle(ctx context.Context, req *Request, buf *Buffer, h UnaryTransportHandler) (*Response, *Buffer, error)
+	Handle(ctx context.Context, req *Request, buf *Buffer, handler UnaryTransportHandler) (*Response, *Buffer, error)
 }
 
 // NopUnaryInboundTransportMiddleware is an inbound middleware that does not do anything special. It
 // simply calls the underlying UnaryTransportHandler.
 var NopUnaryInboundTransportMiddleware UnaryInboundTransportMiddleware = nopUnaryInboundTransportMiddleware{}
 
+type nopUnaryInboundTransportMiddleware struct{}
+
+func (nopUnaryInboundTransportMiddleware) Handle(ctx context.Context, req *Request, buf *Buffer, handler UnaryTransportHandler) (*Response, *Buffer, error) {
+	return handler.Handle(ctx, req, buf)
+}
+
+type unaryTransportHandlerWithMiddleware struct {
+	h UnaryTransportHandler
+	i UnaryInboundTransportMiddleware
+}
+
+func (h unaryTransportHandlerWithMiddleware) Handle(ctx context.Context, req *Request, buf *Buffer) (*Response, *Buffer, error) {
+	return h.i.Handle(ctx, req, buf, h.h)
+}
+
 // ApplyUnaryInboundTransportMiddleware applies the given InboundMiddleware to the given UnaryTransportHandler.
-func ApplyUnaryInboundTransportMiddleware(h UnaryTransportHandler, i UnaryInboundTransportMiddleware) UnaryTransportHandler {
-	if i == nil {
-		return h
+func ApplyUnaryInboundTransportMiddleware(handler UnaryTransportHandler, middleware UnaryInboundTransportMiddleware) UnaryTransportHandler {
+	if middleware == nil {
+		return handler
 	}
-	return unaryTransportHandlerWithMiddleware{h: h, i: i}
+	return unaryTransportHandlerWithMiddleware{h: handler, i: middleware}
 }
 
 // UnaryInboundTransportMiddlewareFunc adapts a function into an InboundMiddleware.
 type UnaryInboundTransportMiddlewareFunc func(context.Context, *Request, *Buffer, UnaryTransportHandler) (*Response, *Buffer, error)
 
 // Handle for UnaryInboundTransportMiddlewareFunc
-func (f UnaryInboundTransportMiddlewareFunc) Handle(ctx context.Context, req *Request, buf *Buffer, h UnaryTransportHandler) (*Response, *Buffer, error) {
-	return f(ctx, req, buf, h)
+func (f UnaryInboundTransportMiddlewareFunc) Handle(ctx context.Context, req *Request, buf *Buffer, handler UnaryTransportHandler) (*Response, *Buffer, error) {
```

Comment:
nit: let's use `reqBuf` instead of just `buf` to be consistent with the rest of spring